azure/storage/__init__.py

```diff
@@ -1279,6 +1286,33 @@ def sign_request(self, request):
         pass
 
 
+class StorageConnectionParameters(object):
+    '''Extract connection parameters from a connection string.
+    
+       This is based on http://azure.microsoft.com/en-us/documentation/articles/storage-configure-connection-string/ .
+       
+       TODO "(Blob|Table|Queue|File)Endpoint" are not yet supported.
+    '''
+    def __init__(self, connection_string = ''):
+        connection_params = dict(s.split('=',1) for s in connection_string.split(';'))
+
+        self.account_name = connection_params.get('AccountName', None)
+        self.account_key = connection_params.get('AccountKey', None)
+        self.protocol = connection_params.get('DefaultEndpointsProtocol', 'https')
+        endpoint_suffix = connection_params.get('EndpointSuffix', None)
+        self.host_base_blob = BLOB_SERVICE_HOST_BASE if endpoint_suffix is None \
+                              else ".blob.{0}".format(endpoint_suffix)
+        self.host_base_table = TABLE_SERVICE_HOST_BASE if endpoint_suffix is None \
+                               else ".table.{0}".format(endpoint_suffix)
+        self.host_base_queue = QUEUE_SERVICE_HOST_BASE if endpoint_suffix is None \
+                               else ".queue.{0}".format(endpoint_suffix)
+        self.dev_host_blob = DEV_BLOB_HOST
+        self.dev_host_table = DEV_TABLE_HOST
+        self.dev_host_queue = DEV_QUEUE_HOST
+        self.timeout = DEFAULT_HTTP_TIMEOUT
+        self.sas_token = connection_params.get('SharedAccessSignature', None)
```

Comment:
Documentation at: http://azure.microsoft.com/en-us/documentation/articles/storage-configure-connection-string/
indicates that the SharedAccessSignature is the base-64 'sig' part of the token.  This won't work if we assign that to sas_token, because sas_token is a partial url query, example:
```
sr=b&sp=r&sv=2014-02-14&st=2015-05-21T00%3A02%3A21Z&sig=bwYazc/DwxJY2nti%2BvDN2voCWlIt%2BgW6f1h8VG%2Bps6o=&se=2015-05-21T01%3A02%3A21Z
```
The comment at the bottom of the page by BScharf appears to confirm that this doesn't work for C# either ;)  For sure, if the user was to try to include the above string in the connection string, it wouldn't get parsed correctly.  And if they just put the 'sig' part, that won't work.

So I would take that line out, and say we don't support SharedAccessSignature in the connection string.

I would also take out the dev_host_X and timeout, since they are not getting initialized from the connection string, and that would prevent the user from setting those values in the dev_host and timeout parameters (they are getting ignored).